docs/architecture/ADR-010-market-params.md

```diff
@@ -0,0 +1,163 @@
+total_supply(token) = total_borrowed(token) + available(token)
+uToken_exchange_rate(token) = total_supply(token) / total_supply(utoken)
+```
+
+### Maximum Supply Utilization
+
+One proposed parameter is `MaxSupplyUtilization`, to be defined per token.
+
+```go
+supply_utilization(token) = total_borrowed(token) / total_supply(token) // ranges 0 - 1
+```
+
+For example, if `1000 ATOM` are supplied and `200 ATOM` are borrowed, `supply_utilization` is `0.2`.
+
+Implementing `MaxSupplyUtilization` would restrict `MsgBorrow` from increasing `total_borrowed` above a desired level.
+
+It may or may not restrict `MsgWithdraw` from decreasing `total_supply(token)` below a desired level - adding this restriction might trap suppliers in a liquidity crisis in exchange for keeping more supply available for `MsgLiquidate`.
+
+| Message Type | Current Decision |
+| - | - |
+| `MsgBorrow` | Restrict |
+| `MsgWithdraw` | Allow |
+| `MsgLiquidate` | Allow |
+
+`MaxSupplyUtilization` could still be indirectly exceeded by borrow interest accruing.
+
+Additionally, dynamic interest rates, which functioned as supply utilization ranged from `0` to `1`, would instead be adjusted expect values from `0` to `MaxSupplyUtilization`.
+
+The motivation for restricting supply utilization is to reduce the likelihood of situations where suppliers cannot use `MsgWithdraw` due to borrowers borrowing all available supply.
+
+This parameter overlaps in function with `MinCollateralLiquidity`. By lowering `MaxSupplyUtilization`, we create a buffer zone where `MsgBorrow` cannot reduce liquidity any further, but `MsgWithdraw` is still available. This helps protect suppliers and makes it more difficult to reach `MinCollateralLiquidity` overall.
+
+### Maximum Collateral Share
+
+Another proposed parameter is `MaxCollateralShare`, to be defined per token.
+
+```go
+total_collateral_value(token) = total_collateral(token) * oracle_price_usd(token)
+collateral_share(token) = total_collateral_value(token) / Sum_All_Tokens(total_collateral_value) // ranges 0 - 1
+```
+
+For example, if `10 ATOM` collateral exists at a price of `$20` alongside `100 OSMO` collateral at a price of `$1`, then `collateral_share(ATOM)` is `10 * $20 / $300 = 0.66` and `collateral_share(OSMO)` is `$100 / $300 = 0.33`.
+
+Implementing `MaxCollateralShare` would restrict `MsgCollateralize` from increasing `total_collateral` above a desired level.
+
+`MaxCollateralShare` could still be indirectly exceeded by fluctuating oracle prices, `MsgDecollateralize` or `MsgLiquidate` of other tokens, or interest accruing in one denom faster than another.
+
+### Minimum Collateral Liquidity
+
+Another proposed parameter is `MinCollateralLiquidity`, to be defined per token.
+
+It is the reciprocal of `MaxCollateralUtilization`, functionally equivalent in which messages it would restrict.
+
+```go
+collateral_liquidity(token) = available(token) / total_collateral(token) // ranges 0 - ∞
```

Comment:
so the definition is the same? Nothing changed.

Reply:
This is proposing using `MinCollateralLiquidity = 0.2` instead of `MaxCollateralUtilization = 5.0`. (Hence collateral utilization appearing in the `# Alternatives` section.

Functionally the same, but `MinCollateralLiquidity` will range 0-1 (helping with basis points), and the naming is different.

It does not aim to address the `1/N` curve for `utilization -> interest rate` as that is unchanged, and out of scope for this design doc.